maintenance/wikia/uncycloUsersMigrator.php

```diff
@@ -0,0 +1,695 @@
+						'user_editcount' => $user->getEditCount(), // use uncyclo counter
+						'user_birthdate' => $user->mBirthDate
+					],
+					$fname
+				);
+
+				$extUser->mId = $dbw->insertId();
+
+				// move user settings to the shared DB
+				$res = $this->getUncycloDB()->select(
+					'user_properties',
+					[ 'up_property', 'up_value' ],
+					array( 'up_user' => $user->getId() ),
+					$fname
+				);
+
+				foreach ( $res as $row ) {
+					$dbw->insert(
+						'user_properties',
+						[
+							'up_user'     => $extUser->mId,
+							'up_property' => $row->up_property,
+							'up_value'    => $row->up_value,
+						],
+						$fname
+					);
+				}
+
+				// mark migrated accounts
+				$dbw->insert(
+					'user_properties',
+					[
+						'up_user'     => $extUser->mId,
+						'up_property' => 'uncyclo_user',
+						'up_value'    => $user->getId(),
+					],
+					$fname
+				);
+			});
+
+			// we have a new ID for a global account
+			// update user ID in uncyclo database
+			$this->doChangeUncycloUserId( $user, $extUser->getId() );
+
+			// delete the user and his settings from the uncyclo DB
+			$this->deleteUncycloUser( $user );
+
+			// invalidate user cache
+			global $wgMemc;
+			$wgMemc->delete( wfMemcKey( 'user', 'id', $user->getId() ) );
+			$wgMemc->delete( wfMemcKey( 'user', 'id', $extUser->getId() ) );
+
+			$dbw->commit(__METHOD__);
+		}
+		catch(Exception $e) {
+			$dbw->rollback();
+			throw $e;
+		}
+
+		$this->info( __METHOD__, [
+			'user'     => $user->getName(),
+			'local_id' => $user->getId(),
+			'ext_id'   => $extUser->getId(),
+		] );
+
+		return $extUser;
+	}
+
+	/**
+	 * Perform the migration of a given uncyclopedia account
+	 *
+	 * @param User $user
+	 */
+	private function migrateUser(User $user) {
+		$this->output( sprintf( "%d: %s <%s>", $user->getId(), $user->getName(), $user->getEmail() ?: 'MISSING_EMAIL' ) );
+
+		// keep data that will be saved to CSV file
+		$action = false;
+		$userName = $user->getName();
+		$isMerged = false;
+
+		// check if the uncyclo account has a valid email set
+		$isValidEmail = Sanitizer::validateEmail( $user->getEmail() );
+		$uncycloEdits = (int) $user->getEditCount();
+		$uncycloEditsSince = $this->getEditsCountAfter( $user, self::UNCYCLO_EDITS_AFTER );
+
+		if ( !$isValidEmail ) {
+			$this->accountsWithNoEmail++;
+
+			if ( $uncycloEditsSince > 0 ) {
+				$this->accountsWithNoEmailWithEdits++;
+
+				$this->output( sprintf( "\n\tEmail missing for %s [active after Jan 2014]\n", $userName ) );
+			}
+		}
+
+		// check the shared users database
+		$globalUser = $this->getGlobalUserByName( $user->getName() );
+		$globalEdits = null;
+
+		if ( $globalUser instanceof User ) {
+			// HACK: calling getEmail() on global user will trigger a SQL query on uncyclo database
+			$this->output( sprintf( ' - conflicts with the global user #%d <%s>', $globalUser->getId(), $globalUser->mEmail ?: 'MISSING_EMAIL' ) );
+
+			$globalEdits  = (int) $globalUser->getEditCount();
```

Comment:
Sure that it fetches number of edits of the global user, not the uncyclo one with the same name or id?